packages/core/src/types.ts

```diff
@@ -467,3 +468,63 @@ export function isReference(value: unknown): value is Reference & { reference: s
  * Global schema singleton.
  */
 export const globalSchema = baseSchema as unknown as IndexedStructureDefinition;
+
+/**
+ * Output the string representation of a value, suitable for use as part of a search query.
+ * @param v The value to format as a string
+ * @returns The stringified value
+ */
+export function formatTypedValue(v: TypedValue): string {
+  switch (v.type) {
+    case PropertyType.uuid:
+    case PropertyType.uri:
+    case PropertyType.url:
+    case PropertyType.string:
+    case PropertyType.oid:
+    case PropertyType.markdown:
+    case PropertyType.id:
+    case PropertyType.code:
+    case PropertyType.canonical:
+    case PropertyType.base64Binary:
+    case PropertyType.SystemString:
+      return v.value as string;
+    case PropertyType.Identifier:
+      return `${v.value.system}|${v.value.value}`;
+    case PropertyType.Coding:
+      return `${v.value.system}|${v.value.code}`;
+    case PropertyType.CodeableConcept:
+      return (v.value as CodeableConcept).coding?.map((c) => `${c.system}|${c.code}`).join(',') ?? v.value.text;
+    case PropertyType.HumanName:
+      if (v.value.text) {
+        return v.value.text;
+      }
+      return `${v.value.given?.join(' ')} ${v.value.family}`;
```

Comment:
Consider using `formatHumanName` in format.ts: https://github.com/medplum/medplum/blob/main/packages/core/src/format.ts#L70